clients/java/src/main/java/io/zeebe/client/impl/util/SocketUtil.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl.util;
+
+import java.net.Inet6Address;
+import java.net.InetSocketAddress;
+
+public final class SocketUtil {
+  private static final String DEFAULT_FORMAT = "%s:%d";
+  private static final String IPV6_FORMAT = "[%s]:%d";
+
+  private SocketUtil() {}
+
+  public static String toHostAndPortString(InetSocketAddress inetSocketAddress) {
+    final String format =
+        inetSocketAddress.getAddress() instanceof Inet6Address ? IPV6_FORMAT : DEFAULT_FORMAT;
```

Comment:
Hm, so I guess if the address is unresolved yet and we return null, it will return as an IPv4 address? Fair enough.

What if I have a domain which resolves to an IPv6 address? Is it going to print `[foobar.com]:80`?

Reply:
Hm, good point! I'll create tests for that :)